test-data/unit/fine-grained.test

```diff
@@ -8673,3 +8673,37 @@ def f(x: Union[int, str]) -> None: ...
 [out]
 ==
 ==
+
+[case testReprocessModuleTopLevelWhileMethodDefinesAttrExplicit]
+import a
+[file a.py]
+from b import B
+
+B().x
+
+[file a.py.3]
+from b import B
+
+x = B().x
+
+[file b.py]
+from c import f
+f(int())
+
+class A:
+    x: int
+
+class B(A):
+    def meth(self) -> None:
+        self.x: int
+
+[file c.py]
+def f(x: int) -> None: ...
+
+[file c.py.2]
+from typing import Union
+def f(x: Union[int, str]) -> None: ...
+
+[out]
```

Comment:
Would it also make sense to test the explicit protocol base class case and the case where we don't have a type annotation but just assign to `self.x`? What about attribute defined in class body of the derived class? These may be covered by other test cases. If so, maybe verify the updated targets in those test cases as well?

Reply:
OK, I will add couple of test cases.